File: Semester 1/Class04_Computer_Contest_Lv_2_Notes_and_Homework_Oct_02-03/lvl2_hw_sol/lvl2_clockwise_fence.py
import sys
input = sys.stdin.readline
cw_turn = ['NE', 'ES', 'SW', 'WN']
ccw_turn = ['NW', 'EN', 'SE', 'WS']
for _ in range(int(input())):
    s = input().strip()
    cw, ccw = 0, 0
    for i in range(len(s)-1):
        sub = s[i:i+2]
        if sub in cw_turn:
            cw+=1
        elif sub in ccw_turn:
            ccw+=1
    print('CW') if cw > ccw else print('CCW')

File: Semester 1/Class04_Computer_Contest_Lv_2_Notes_and_Homework_Oct_02-03/lvl2_hw_sol/lvl2_clockwise_fence.java
import java.util.*;
import java.io.*;
public class Main {
	static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	static PrintWriter pr = new PrintWriter(new BufferedWriter(new OutputStreamWriter(System.out)));
	static StringTokenizer st;
	public static void main(String[] args) throws IOException{	
		int n = readInt();
		for(int t=0; t<n; t++) {
			String s = readLine();
			int cw = 0, ccw = 0;
			for(int i=0; i<s.length()-1; i++) {
				String sub = s.substring(i, i+2); //get [i, i+1]
				if(sub.equals("NE") || sub.equals("ES") || sub.equals("SW") || sub.equals("WN")) {
					cw++;
				}
				if(sub.equals("NW") || sub.equals("EN") || sub.equals("SE") || sub.equals("WS")) {
					ccw++;
				}				
			}
			if(cw > ccw) System.out.println("CW");
			else System.out.println("CCW");
		}
	}
	static String next () throws IOException {
		while (st == null || !st.hasMoreTokens())
			st = new StringTokenizer(br.readLine().trim());
		return st.nextToken();
	}
	static long readLong () throws IOException {
		return Long.parseLong(next());
	}
	static int readInt () throws IOException {
		return Integer.parseInt(next());
	}
	static double readDouble () throws IOException {
		return Double.parseDouble(next());
	}
	static char readCharacter () throws IOException {
		return next().charAt(0);
	}
	static String readLine () throws IOException {
		return br.readLine().trim();
	}
}